src/distilabel/steps/tasks/sentence_transformers.py

```diff
@@ -0,0 +1,126 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+import sys
+from typing import TYPE_CHECKING, Any, Dict, Final, List, Literal, Optional, Union
+
+from jinja2 import Template
+
+from distilabel.steps.tasks.base import Task
+
+if sys.version_info < (3, 9):
+    import importlib_resources
+else:
+    import importlib.resources as importlib_resources
+
+if TYPE_CHECKING:
+    from distilabel.steps.tasks.typing import ChatType
+
+GenerationAction = Literal["paraphrase", "semantically-similar", "query"]
+
+POSITIVE_NEGATIVE_PAIR_REGEX = re.compile(
+    r"## Positive\s+(.*?)\s+(?:## Negative\s+(.*?)\s*)?$",
+    re.DOTALL,
+)
+
+
+GENERATION_ACTION_SENTENCES: Final[Dict[GenerationAction, str]] = {
+    "paraphrase": "paraphrase",
+    "semantically-similar": "be semantically similar to",
+    "query": "be a query for",
+}
+
+
+class GenerateSentencePair(Task):
+    """Generate a positive and negative (optionally) sentences given an anchor sentence.
+
+    `GenerateSentencePair` is a pre-defined task that given an anchor sentence generates
+    a positive sentence related to the anchor and optionally a negative sentence unrelated
+    to the anchor. This task is useful to generate training datasets for training embeddings
+    models.
+
+    Attributes:
+        triplet: a flag to indicate if the task should generate a triplet of sentences
+            (anchor, positive, negative). Defaults to `False`.
+        action: the action to perform to generate the positive sentence.
+
+    Input columns:
+        - anchor (`str`): The anchor sentence to generate the positive and negative sentences.
+
+    Output columns:
+        - positive (`str`): The positive sentence related to the `anchor`.
+        - negative (`str`): The negative sentence unrelated to the `anchor`.
+
+    Categories:
+        - embedding
+    """
+
+    triplet: bool = False
+    action: GenerationAction
+
+    def load(self) -> None:
+        """Loads the Jinja2 template."""
+        super().load()
+
+        _path = str(
+            importlib_resources.files("distilabel")
+            / "steps"
+            / "tasks"
+            / "templates"
+            / "generate-sentence-pair.jinja2"
+        )
+
+        self._template = Template(open(_path).read())
+
+    @property
+    def inputs(self) -> List[str]:
+        return ["anchor"]
+
+    def format_input(self, input: Dict[str, Any]) -> "ChatType":
+        return [
+            {
+                "role": "user",
+                "content": self._template.render(
+                    anchor=input["anchor"],
+                    action=GENERATION_ACTION_SENTENCES[self.action],
+                ),
+            }
+        ]
```

Comment:
Did you try with a system prompt? As of previous prompt-engineering attempts, I've seen that adding a system prompt works really well in these scenarios